app/controllers/api/v3/stream_daily_averages_controller.rb

```diff
@@ -0,0 +1,16 @@
+module Api
+  module V3
+    class StreamDailyAveragesController < BaseController
+      def show
+        averages = StreamDailyAveragesRepository.new.from_time_range(stream_id: params[:id], start_date: params[:start_date], end_date: params[:end_date])
+        serialized_averages = StreamDailyAveragesSerializer.new.call(averages)
+
+        if serialized_averages.present?
+          render json: serialized_averages, status: :ok
+        else
+          render json: { error: 'No data available' }, status: :not_found
```

Comment:
Not sure about cases when there is no data - should it empty array or an error? 🤔